xapian-core/tests/harness/backendmanager_honey.cc

```diff
@@ -88,6 +88,28 @@ BackendManagerHoney::get_alt_writable_database(const string & name,
     return wdb;
 }
 
+Xapian::WritableDatabase
+BackendManagerHoney::get_generated_database(const string& name)
+{
+    return get_alt_writable_database(name, string());
+}
+
+void
+BackendManagerHoney::finalise_generated_database(const string& path)
+{
+    // converting a glass backend to honey
+    string tmpfile = path;
+    tmpfile += ".tmp";
```

Comment:
I think it would be better to pass in `tmpfile` to this method - as this is the knowledge about how the temporary file is named has to be in duplicated in (at least) two different files.

Reply:
As discussed on IRC, I was confusing this temporary file with a different one.  There is a problem I think, but not what I thought it was.

If I follow, for honey we:

* call get_generated_database() writing a glass database to a temporary path
* then we rename that to the final path (this is bad - if the testsuite is interrupted at this point we leave a valid glass database in the honey database cache)
* then we compact that and convert it to honey at a temporary path
* then we remove the glass database from the final path and rename the honey one there instead

We really want to skip the second step and compact and convert to honey from the first temporary path to a second temporary path instead.

An alternative would be to lazily generate the glass database to a temporary path and then rename it to the final name in the **glass** database cache and then compact and convert from there to a temporary path for honey which then gets renamed to the final name in the honey cache.  That would save work as we avoid an extra generation of the glass database corresponding to each generated honey database, but probably means more scheduling restrictions for the parallel stuff later.

Reply:
Made the required changes.